ckan/logic/action/create.py

```diff
@@ -1062,12 +1062,12 @@ def user_invite(context: Context,
     data['name'] = name
     # send the proper schema when creating a user from here
     # so the password field would be ignored.
-    context['schema'] = ckan.logic.schema.create_user_for_user_invite_schema()
+    invite_schema = ckan.logic.schema.create_user_for_user_invite_schema()
+    
     data['state'] = model.State.PENDING
-    user_dict = _get_action('user_create')(context, data)
-    # we already created the user so we don't need the schema anymore
-    # as we are calling for other actions bellow
-    context.pop('schema')
+    user_dict = _get_action('user_create')(
+        cast(Context, dict(context, schema=invite_schema)),
```

Comment:
@smotornyuk I'm still learning about best practices with python typing, would writing this as `Context(context, schema=invite_schema)` be better than using `cast` here?

Reply:
Yes, exactly. Your option is preferred all the time. `cast` can be used only when we have no other options. But due to the nature of the `Context`(`TypedDict`), the following form must be used(with no positional args):
```
Context(**context, **{"schema": invite_schema})
```

For example `Context(model=ckan.model)` will likely fail, because at the moment typing libraries do not work well with python modules passed around. And we have to use cast in such a case(eventually, this also will be allowed and `cast` will no longer be needed). 
Almost everywhere else, `Context(**context)` will do the thing.

Reply:
It's a fun part.

`TypedDict` is substituted by the `dict` in runtime. In this way, all the operations from the `dict` are allowed, including positional arguments to the constructor.

But technically, `TypedDict` is a `Mapping`, which does not define a constructor with the positional args. So, while your example does not produce an error as a python code, it'll likely report an invalid operation if you check it with `pyright`

Reply:
I think you're saying it works but pyright doesn't understand it so we have to make our code uglier to pass the typing tests we added. That makes me sad.

Reply:
Yes, if we put it straight, that's the reason.

Reply:
But I'll create a feature request in the pyright source. Maybe, even if the authors won't change their mind, they'll suggest more coinse way to deal with our case

Reply:
But, there is a [PEP-589](https://peps.python.org/pep-0589/) which clearly says:

> It can be used as a callable object with keyword arguments corresponding to the TypedDict items. Non-keyword arguments are not allowed.

So this ugly code is a part of python's design :)